MediaGalleryUi/Model/SearchCriteria/CollectionProcessor/JoinProcessor/BlockId.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MediaGalleryUi\Model\SearchCriteria\CollectionProcessor\JoinProcessor;
+
+use Magento\Framework\Api\SearchCriteria\CollectionProcessor\JoinProcessor\CustomJoinInterface;
+use Magento\Framework\App\ResourceConnection;
+use Magento\Framework\Data\Collection\AbstractDb;
+/**
+ * Used to joint table of the collection based on the filter selection.
+ */
+class BlockId implements CustomJoinInterface
```

Comment:
Let's removed this join and use a similar approach from the one used on the `Used In` filter